src/pycram/designators/action_designator.py

```diff
@@ -294,8 +294,72 @@ class Action(ActionDesignatorDescription.Action):
 
         @with_tree
         def perform(self) -> None:
+            # Store the object's data copy at execution
             self.object_at_execution = self.object_designator.data_copy()
-            PickUpMotion(object_desig=self.object_designator, arm=self.arm, grasp=self.grasp).resolve().perform()
+            robot = BulletWorld.robot
+            # Retrieve object and robot from designators
+            object = self.object_designator.bullet_world_object
+            # Get grasp orientation and target pose
+            grasp = robot_description.grasps.get_orientation_for_grasp(self.grasp)
+
+            # oTm = Object Pose in Frame map
+            oTm = object.get_pose()
+            # Transform the object pose to the map frame
+            mTo = object.local_transformer.transform_to_object_frame(oTm, object)
+            # Adjust the pose according to the special knowledge of the object designator
+            adjusted_pose = self.object_designator.special_knowledge_adjustment_pose(self.grasp, mTo)
+            # Transform the adjusted pose to the map frame
+            adjusted_oTm = object.local_transformer.transform_pose(adjusted_pose, "map")
+            # multiplying the orientation therefore "rotating" it, to get the correct orientation of the gripper
+            ori = helper.multiply_quaternions([adjusted_oTm.orientation.x, adjusted_oTm.orientation.y,
+                                        adjusted_oTm.orientation.z, adjusted_oTm.orientation.w],
+                                       grasp)
+
+            # Set the orientation of the object pose by grasp in MAP
+            adjusted_oTm.orientation.x = ori[0]
+            adjusted_oTm.orientation.y = ori[1]
+            adjusted_oTm.orientation.z = ori[2]
+            adjusted_oTm.orientation.w = ori[3]
+
+            # prepose depending on the gripper (its annoying we have to put pr2_1 here tbh
+            gripper_frame = "pr2_1/l_gripper_tool_frame" if self.arm == "left" else "pr2_1/r_gripper_tool_frame"
```

Comment:
This would only work in very specific cases. Please use: robot.get_link_tf_frame(robot_description.get_tool_frame(self.arm))